component/grant-type/src/main/java/org/wso2/carbon/identity/oauth2/grant/token/exchange/TokenExchangeGrantHandler.java

```diff
@@ -0,0 +1,272 @@
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static org.wso2.carbon.identity.oauth2.grant.token.exchange.utils.TokenExchangeUtils.handleException;
+
+/**
+ * Class to handle Token Exchange grant type
+ */
+public class TokenExchangeGrantHandler extends AbstractAuthorizationGrantHandler {
+
+    private static final Log log = LogFactory.getLog(TokenExchangeGrantHandler.class);
+    private int validityPeriod;
+    private boolean validateIAT = true;
+    private String[] registeredClaimNames = new String[]{"iss", "sub", "aud", "exp", "nbf", "iat", "jti"};
+    private String requested_token_type = TokenExchangeConstants.JWT_TOKEN_TYPE;
+
+    @Override
+    public void init() throws IdentityOAuth2Exception {
+        super.init();
+
+        /*
+          From identity.xml following configs are read.
+
+          <OAuth>
+              <TokenExchangeGrant>
+                  <EnableIATValidation>true</EnableIATValidation>
+                  <IATValidityPeriod>30</IATValidityPeriod>
+              </TokenExchangeGrant>
+          </OAuth>
+         */
+
+        String validateIATProp = IdentityUtil.getProperty(TokenExchangeConstants.PROP_ENABLE_IAT_VALIDATION);
+        if (StringUtils.isNotBlank(validateIATProp)) {
+            validateIAT = Boolean.parseBoolean(validateIATProp);
+        }
+
+        String validityPeriodProp = IdentityUtil.getProperty(TokenExchangeConstants.PROP_IAT_VALIDITY_PERIOD);
+
+        if (validateIAT) {
+            if (StringUtils.isNotBlank(validityPeriodProp)) {
+                try {
+                    validityPeriod = Integer.parseInt(validityPeriodProp);
+                } catch (NumberFormatException e) {
+                    validityPeriod = TokenExchangeConstants.DEFAULT_IAT_VALIDITY_PERIOD;
+                    log.warn("Invalid value: " + validityPeriodProp + " is set for IAT validity period. Using " +
+                            "default value: " + validityPeriod + " minutes.");
+                }
+            } else {
+                validityPeriod = TokenExchangeConstants.DEFAULT_IAT_VALIDITY_PERIOD;
+                log.warn("Empty value is set for IAT validity period. Using default value: " + validityPeriod
+                        + " minutes.");
+            }
+        }
+
+        String registeredClaims = IdentityUtil.getProperty(TokenExchangeConstants.REGISTERED_CLAIMS);
+        if (StringUtils.isNotBlank(registeredClaims)) {
+            registeredClaimNames = registeredClaims.split("\\s*,\\s*");
+        }
+
+        if (log.isDebugEnabled()) {
+            log.debug("Validate IAT is set to: " + validateIAT + " for JWT grant.");
+            if (validateIAT) {
+                log.debug("IAT validity period is set to: " + validityPeriod + " minutes for JWT grant.");
+            }
+        }
+    }
+
+    @Override
+    public boolean validateGrant(OAuthTokenReqMessageContext tokReqMsgCtx) throws IdentityOAuth2Exception {
+        SignedJWT signedJWT;
```

Comment:
add a new line

Reply:
Fixed in 8693d5454cb05015e9d9614c56a1d09bdf8a8997